From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 4 Apr 2012 05:32:11 +0100
Subject: [PATCH] Revert "rtc: Provide flag for rtc devices that don't support
 UIE"

This reverts commit 61d38cd909ad35ff497bf1a84cd57b2baf6d9c49.
It changes ABI and is only needed for rtc-mpc5121 which we don't
enable.
---
 drivers/rtc/interface.c   |    5 -----
 drivers/rtc/rtc-mpc5121.c |    2 --
 include/linux/rtc.h       |    3 +--
 3 files changed, 1 insertion(+), 9 deletions(-)

diff --git a/drivers/rtc/interface.c b/drivers/rtc/interface.c
index 565742b..c1edbf8 100644
--- a/drivers/rtc/interface.c
+++ b/drivers/rtc/interface.c
@@ -445,11 +445,6 @@ int rtc_update_irq_enable(struct rtc_device *rtc, unsigned int enabled)
 	if (rtc->uie_rtctimer.enabled == enabled)
 		goto out;
 
-	if (rtc->uie_unsupported) {
-		err = -EINVAL;
-		goto out;
-	}
-
 	if (enabled) {
 		struct rtc_time tm;
 		ktime_t now, onesec;
diff --git a/drivers/rtc/rtc-mpc5121.c b/drivers/rtc/rtc-mpc5121.c
index 0fc2d22..da60915 100644
--- a/drivers/rtc/rtc-mpc5121.c
+++ b/drivers/rtc/rtc-mpc5121.c
@@ -360,8 +360,6 @@ static int __devinit mpc5121_rtc_probe(struct platform_device *op)
 						&mpc5200_rtc_ops, THIS_MODULE);
 	}
 
-	rtc->rtc->uie_unsupported = 1;
-
 	if (IS_ERR(rtc->rtc)) {
 		err = PTR_ERR(rtc->rtc);
 		goto out_free_irq;
diff --git a/include/linux/rtc.h b/include/linux/rtc.h
index fcabfb4..93f4d03 100644
--- a/include/linux/rtc.h
+++ b/include/linux/rtc.h
@@ -202,8 +202,7 @@ struct rtc_device
 	struct hrtimer pie_timer; /* sub second exp, so needs hrtimer */
 	int pie_enabled;
 	struct work_struct irqwork;
-	/* Some hardware can't support UIE mode */
-	int uie_unsupported;
+
 
 #ifdef CONFIG_RTC_INTF_DEV_UIE_EMUL
 	struct work_struct uie_task;
-- 
1.7.9.5

